### PR TITLE
Add inline doc comments to the Dart and Flutter SDKs

### PR DESCRIPTION
**Pull request by stnguyen90**
## What does this PR do?

Add docs that will appear in IDEs and on pub.dev.

## Test Plan

Manually generated SDKs:

* https://github.com/appwrite/sdk-for-flutter/compare/feat-152-doc-comments

## Related PRs and Issues

* https://github.com/appwrite/sdk-for-flutter/issues/152
* https://github.com/appwrite/sdk-for-flutter/compare/feat-152-doc-comments

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: templates/dart/lib/permission.dart.twig [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
+
+  /// Write permission for provided [role]
+  ///
+  /// This is an alias of update, delete, and possibly create.
+  /// Don't use write in combination with update, delete, or create.
+  static String write(String role) {
+    return 'write("$role")';
+  }
+
+  /// Update permission for provided [role]
```
Comment (lohanidamodar) [inline templates/dart/lib/permission.dart.twig:20]:
```suggestion
  /// Create permission for provided [role]
```

File: templates/dart/lib/query.dart.twig [hunk 2]
```diff
@@ -1,61 +1,97 @@
...
+/// Helper class to generate query strings.
 class Query {
-    Query._();
-    
+  Query._();
+
+  /// Filter resources where [attribute] is equal to [value].
+  /// 
+  /// [value] can be a single value or a list. If a list is used
+  /// the query will return resources where [attribute] is equal
```
Comment (lohanidamodar) [inline templates/dart/lib/query.dart.twig:10]: this part is suitable for many other filters, let's update all relevant

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Awesome looking great. Couple of typos to correct.

File: templates/dart/lib/query.dart.twig (hunk 2 above)
Comment (stnguyen90) [reply templates/dart/lib/query.dart.twig:10]: You mean for the other sdks? You want me to add it for all of them now? Would it be quicker to get this in first and then work on the other soon after?

File: templates/dart/lib/permission.dart.twig (hunk 1 above)
Comment (stnguyen90) [reply templates/dart/lib/permission.dart.twig]: @stnguyen90 updated.

Comment (stnguyen90) [reply templates/dart/lib/permission.dart.twig]: @lohanidamodar, I've fixed this typo.

File: templates/dart/lib/query.dart.twig (hunk 2 above)
Comment (lohanidamodar) [reply templates/dart/lib/query.dart.twig:10]: I mean for other methods in this query file

**Review by lohanidamodar — APPROVED** [review]

**Pull request merged**